usr/src/uts/common/fs/zfs/dsl_dataset.c

```diff
@@ -883,14 +884,41 @@ dsl_dataset_create_sync_dd(dsl_dir_t *dd, dsl_dataset_t *origin,
 	return (dsobj);
 }
 
+static int
+deadlist_enqueue_cb(void *arg, const blkptr_t *bp, dmu_tx_t *tx)
```

Comment:
@ahrens one question: should `dmu_buf_rele` belong to `dsl_dataset_sync_done` or should it be external to it? Because the extra reference is not needed in the `dsl_dataset_zero_zil` case.